pkg/lib/cockpit.js

```diff
@@ -1483,17 +1483,27 @@ function factory() {
             });
     };
 
-    function format_units(number, suffixes, factor, options) {
-        // backwards compat: "options" argument position used to be a boolean flag "separate"
-        if (!is_object(options))
-            options = { separate: options };
+    function format_units(suffixes, number, second_arg, third_arg) {
+        let options = second_arg;
 
+        /* compat API: we used to accept 'factor' as a separate second arg */
+        if (third_arg || (second_arg && !is_object(second_arg))) {
+            options = third_arg;
+
+            // double backwards compat: "options" argument position used to be a boolean flag "separate"
+            if (!is_object(options))
+                options = { separate: options };
+
+            options.factor = second_arg;
+        }
+
+        const factor = options?.factor || (options?.base2 ? 1024 : 1000);
         let suffix = null;
 
         /* Find that factor string */
         if (!number && number !== 0) {
             suffix = null;
-        } else if (typeof (factor) === "string") {
+        } else if (typeof factor === "string") {
```

Comment:
FTR: I'd love to kill this with fire. Requires https://github.com/cockpit-project/cockpit-podman/pull/1673 and the corresponding change to the overview page.

Reply:
I'm not against ripping this out, but:
 - I want to be very sure nobody is going to be hurt
 - I want it in a different PR

Reply:
Right, this was just an angry drive-by comment, not actionable here :grin: